### ISSUE TITLE
Handle migration of fields with a reserved word as a handle more gracefully.

### ISSUE THREAD

**Issue by jdsdev** [verbb/formie#568]
**What are you trying to do?**

When a reserved word is used as a field handle in Freeform, the field fails to migrate to Formie, this propagates to all submissions for that form.

**What's your proposed solution?**

In my case, simply prefixing the reserved handles and then migrating the content would work perfectly. For people who _need_ to have the same handles, perhaps provide the option to choose whether to prefix or fail on reserved handles.

Additionally, after the migration an error is shown for every submission that has a field that failed to migrate, making the migration log quite messy for these forms. It would be preferable to either hide or minimize the full error log.

**Comment by engram-design** [comment]
Yep, great point. Improved for the next release, where fields are automatically prefixed with `field_`. I think having the option to opt-in to the behaviour (fail) is probably not great, as we'll want to migrate all the content first, raise a warning that the field has been renamed, and then they can at least rename it later in the form builder if they want a specific handle.

To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.15",
  "...": "..."
}
```

Then run `composer update`.

**Comment by jdsdev** [comment]
Thanks for the update, it doesn't seem to handle all cases yet though. There are two issues I'm still encountering:

1) `in_array` is case sensitive, and the later `->validate()` is not, so reserved handles are still slipping through.

2) Handles aren't being updated for form submissions, so they're still throwing errors.

**Comment by engram-design** [comment]
Ah, great point about case sensitivity with handles. Both those items should be updated in the `craft-3` branch, so run another `composer update`.

**Comment by jdsdev** [comment]
Thanks for all of the quick fixes. The migration log is much more digestible now, and I just see a couple other issues.

1) Some of the form and field handles have characters considered illegal by Formie, namely "-" in my case. Perhaps just strip them before migrating?

2) HTML fields are failing to migrate. Without digging further at the moment, the log is showing the following error:

```
labelPosition: Label Position is invalid
```

**Comment by engram-design** [comment]
Thanks for your persistence here. I didn't actually think fields with a `-` would save, even in Freeform, but I suppose they have their own non-standard handle validation. These items are also fixed in the `craft-3` branch.

**Comment by engram-design** [comment]
Fixed in [1.4.16](https://github.com/verbb/formie/releases/tag/1.4.16)
